src/plugins/validate-semantic/validators/oas3/refs.js

```diff
@@ -6,26 +6,58 @@ export const validateOAS3RefsForRequestBodiesReferenceRequestBodyPositions = ()
         const value = node.node
         const ref = value.$ref
 
-        if(!ref) {
+        if (!ref) {
           return acc
         }
 
         const [, refPath = ""] = ref.split("#")
         const pathArr = refPath.split("/") || []
         const parentRefKey = pathArr.slice(-2)[0]
         const targetRefKey = pathArr.slice(-1)[0]
-        if(
-          targetRefKey !== "requestBody"
-          && parentRefKey !== "requestBody"
-          && parentRefKey !== "requestBodies"
-        ) {
+
+        // External references starting with http:// or https://
+        if (ref.startsWith("https://") || ref.startsWith("http://")){
+          return acc  
+        }
+
+        // External references starting with ./
+        if (ref.startsWith("./")){
+          return acc  
+        }
+
+        // Extensions are valid
+        if (ref.startsWith("#/") && (parentRefKey.startsWith("/x-") || targetRefKey.startsWith("/x-"))){
```

Comment:
@ralphdoe 
1) `startsWith("/x-")` -->`startsWith("x-")`  (i.e. no slash)
2) All items of `pathArr` need to be checked for `x-`, not just `parentRefKey` and `targetRefKey`. The logic should be
```
if local $ref
and
any item in pathArr starts with "x-"
```

Reply:
Done.